packages/server/__tests__/controllers/guilds.test.js

```diff
@@ -0,0 +1,201 @@
+    banner: "URL to Banner Image",
+    icon: "URL to Icon Image",
+    media: ["Media1", "Media2", "Media3"],
+    isInviteOnly: true,
+  };
+
+  const expectedUpdateGuildResponse = {
+    guildId: "308b5bc4-7771-4514-99b7-9611cefb7e1d",
+    name: "Apple",
+    handler: "Tim Cook",
+    description: "Come buy our overpriced iphones.",
+    category: "Guild Category",
+    location: "Optional Location",
+    website: "Optional Website",
+    tags: ["Tag1", "Tag2", "apple 15"],
+    banner: "URL to Banner Image",
+    icon: "URL to Icon Image",
+    media: ["Media1", "Media2", "Media3"],
+    isInviteOnly: true,
+  };
+
+  const initialMultipleUpdateGuildResponseOneField = {
+    name: "Facebook",
+  };
+
+  const expectedUpdateGuildResponseOneField = {
+    guildId: "308b5bc4-7771-4514-99b7-9611cefb7e1d",
+    name: "Facebook",
+  };
+
+  test("should update multiple guild properties successfully", async () => {
+    prismaMock.guilds.update.mockResolvedValue(expectedUpdateGuildResponse);
+
+    const result = await GuildsController.updateGuild(
+      "308b5bc4-7771-4514-99b7-6666cefb7e1d",
+      initialMultipleUpdateGuildResponse
+    );
+
+    expect(result).toEqual(
+      expect.objectContaining(expectedUpdateGuildResponse)
+    );
+  });
+
+  test("should update one guild property successfully", async () => {
+    prismaMock.guilds.update.mockResolvedValue(
+      expectedUpdateGuildResponseOneField
+    );
+
+    const result = await GuildsController.updateGuild(
+      "308b5bc4-7771-4514-99b7-6666cefb7e1d",
+      initialMultipleUpdateGuildResponseOneField
+    );
+
+    expect(result).toEqual(
+      expect.objectContaining(expectedUpdateGuildResponseOneField)
+    );
+  });
+
+  // test("should fail attempting to update an nonexistent guild property", async () => {
+  //   prismaMock.guilds.update.mockResolvedValue(
+  //     new PrismaClientKnownRequestError("Missing required field error", {code: "P2012"})
+  //   );
+
+  //   await expect(
+  //     GuildsController.updateGuild(testGuild)).resolves.toThrow(
+  //       PrismaClientKnownRequestError
+  //     )
+  // });
+
+  // test("should fail attempting to update an nonexistent guild", async () => {
+  //   prismaMock.guilds.update.mockResolvedValue(
+  //     new PrismaClientKnownRequestError("Missing required field error", {code: "P2012"})
+  //   );
+
+  //   await expect(
+  //     GuildsController.updateGuild(testGuild)).resolves.toThrow(
+  //       PrismaClientKnownRequestError
+  //     )
+  // });
+
+  /*
+  Unit Tests: Deleting Guilds
+  - Successfully deleting guild by ID
+  - Trying to delete a nonexistent guild (should throw error)
+  */
+
+  const guildIdToDelete = "308b5bc4-7771-4514-99b7-6666cefb7e1d";
+
+  test("should successfully delete a target guild", async () => {
+    prismaMock.guilds.delete.mockResolvedValue(guildIdToDelete);
+
+    const result = await GuildsController.deleteGuild(guildIdToDelete);
+
+    expect(result).toEqual(guildIdToDelete);
+  });
```

Comment:
According to Prisma docs, when deleting from the database, the `delete()` query returns the object that was deleted. Change your mock resolved value to a mock guild object instead.

![image](https://github.com/cppsea/icebreak/assets/36950420/2ccc537d-4e8d-4cd7-a00b-ac99fc5f9616)